---
title: "Homework 08"
author: "Biology Student"
date: "20 Apr 2020"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Objectives

For this assignment you will need to characterize several attributes of a differential expression transcriptomics study and compare them to the study that you are using for Project 4.
- [ ] Differential Expression experimental design
- [ ] Sample replication
- [ ] Sequencing depth and format
- [ ] Quality control steps performed
- [ ] Mapping program used
- [ ] DE analysis performed

## Example paper

RNA-Seq performed by:
Frahm KA, Waldman JK, Luthra S, Rudine AC, Monaghan-Nichols AP, Chandran UR. A comparison of the sexually dimorphic dexamethasone transcriptome in mouse cerebral cortical and hypothalamic embryonic neural stem cells. Mol Cell Endocrinol. 2017; https://doi.org/10.1016/j.mce.2017.05.026.

You can find a copy of this paper in the Readings folder. 

Abstract
Fetal exposure to synthetic glucocorticoids reprograms distinct neural circuits in the developing brain, often in a sex-specific manner, via mechanisms that remain poorly understood. To reveal whether such reprogramming is associated with select molecular signatures, we characterized the transcriptome of primary, embryonic mouse cerebral cortical and hypothalamic neural progenitor/stem cells derived from individual male and female embryos exposed to the synthetic glucocorticoid, dexamethasone. Gene expression profiling by RNA-Seq identified differential expression of common and unique genes based upon brain region, sex, and/or dexamethasone exposure. These gene expression datasets provide a unique resource that will inform future studies examining the molecular mechanisms responsible for region- and sex-specific reprogramming of the fetal brain brought about by in utero exposure to excess glucocorticoids.

Link to the dataset: https://www.ncbi.nlm.nih.gov/sra/SRP100701

## Your paper

Summarize your paper here. Provide links to the paper and to the RNAseq dataset.

## Questions

Answer these questions for each paper. Note that it is okay (well, okay for you) if this information is not available in the paper. In that case, indicate *NOT FOUND*.

1. Differential Expression experimental design
What was the organism, tissue, and treatment groups? Was there anything special about the RNA isolation procedures?

2. Sample replication
Were RNA samples pooled? How many individuals were used for each treatment group? What criteria was used to determine if this sample size was sufficient?

3. Sequencing depth and format
What was the depth and type of sequencing performed? What criteria was used to determine if this sequencing depth was sufficient?

4. Quality control steps performed
How was the data cleaned (trimmed) and checked for quality? Were any samples excluded (and, if so, why)?

5. Mapping program used
Finally, an easy question.

6. DE analysis performed
What program was used? What cutoff values were used? Was more than one DE program used? (If so, how were results presented?)

